### ISSUE TITLE
Reading using write permissions

### ISSUE THREAD

**Issue by nighelles** [rethinkdb/rethinkdb#5599]
@VeXocide 

```
r.table("open").insert({id:1, super_secret_thing: "Applesauce"})
r.table("open").insert([{"id":i} for i in range(0,10)], return_changes="always").run()

{'inserted': 9, 'unchanged': 0, 'errors': 1, 'replaced': 0, 'deleted': 0, 'skipped': 0, 'changes': [ ... 'error': 'Duplicate primary key `id`:\n{\n\t"id":\t1,\n\t"super_secret_thing":\t"Applesauce"\n} ...}
```

**Comment by VeXocide** [comment]
Nice find, after some discussion we've decided to require read permissions when writing as well.

**Comment by VeXocide** [comment]
Fixed in `next` via commit 807da35498c9c4c7b64f3c35e78ca571ee0fe275, over-the-shoulder by @danielmewes.
